src/js/searchScript.js

```diff
@@ -0,0 +1,82 @@
+const projectList = [
+  {
+    src: "spring-boot.svg",
+    title: "Spring Boot",
+    text: "Takes an opinionated view of building Spring applications and gets you up and running as quickly as possible.",
+  },
+  {
+    src: "spring-framework.svg",
+    title: "Spring Framework",
+    text: "Provides core support for dependency injection, transaction management, web apps, data access, messaging, and more.",
+  },
+  {
+    src: "spring-data.svg",
+    title: "Spring Data",
+    text: "Provides a consistent approach to data access – relational, non-relational, map-reduce, and beyond.",
+  },
+  {
+    src: "spring-cloud.svg",
+    title: "Spring Cloud",
+    text: "Provides a set of tools for common patterns in distributed systems. Useful for building and deploying microservices.",
+  },
+  {
+    src: "spring-data-flow.svg",
+    title: "Spring Cloud Data Flow",
+    text: "Provides an orchestration service for composable data microservice applications on modern runtimes.",
+  },
+  {
+    src: "spring-security.svg",
+    title: "Spring Security",
+    text: "Protects your application with comprehensive and extensible authentication and authorization support.",
+  },
+];
+
+insertResult(projectList);
+
+let typing;
+document
+  .getElementById("searchInput")
+  .addEventListener("keyup", function (event) {
+    clearTimeout(typing);
```

Comment:
It is very good that you already know the right way to implement search funnctionallity
FYI there are already defined patterns for such cases, you can read about `debounce` function
https://www.youtube.com/watch?v=TBIEArmPywU